File: typescript-project-euler-100/src/p001.ts
import { Solution } from "./Solution";

export class P001 extends Solution{
    solve() {
        this.problemNumber = 1;
        const last = 1000;
        let sum = 0;

        for (let i = 1; i < last; i++) {
            if ((i % 3 == 0) || (i % 5 == 0)) {
                sum = sum + i;
            }
        }
        return sum;
    }
}

File: csharp-project-euler-100/src/P001.cs


namespace net.talaatharb.projecteuler100
{

    /**
    * @author mharb
    *
    */
    public class P001 : Solution
    {

        /**
         * Problem solution entry point
         * 
         * @param args
         */
        public static void Main(string[] args)
        {
            new P001().run();
        }

        public override long solve()
        {
            problemNumber = 1;
            const int last = 1000;
            int sum = 0;

            for (int i = 1; i < last; i++)
            {
                if ((i % 3 == 0) || (i % 5 == 0))
                {
                    sum = sum + i;
                }
            }

            return sum;
        }
    }
}

